src/converter/baseConverter.ts

```diff
@@ -0,0 +1,37 @@
+import { validationMessage } from '@/constants/validationMessages';
+import type { ResultWithValue } from '@/contracts/result';
+import type { ValidationResult } from '@/contracts/validationResult';
+
+interface IBaseConverter<TI, TO> {
+  input: TI;
+  converter: (input: TI) => TO;
+  inputValidator: (input: TI) => ValidationResult;
+}
+
+export const baseConverter =
+  <TI, TO extends {}>(props: IBaseConverter<TI, TO>) =>
+  (): ResultWithValue<TO> => {
+    const validationResult = props.inputValidator(props.input);
+    if (validationResult.isValid === false) {
```

Comment:
Yup it would, I like the strict comparison over the `!`. I could be convinced to change it, but then we should try find a prettier rule to enforce it 😋